sql/persisted_variable.cc

```diff
@@ -768,13 +792,29 @@ bool Persisted_variables_cache::set_persist_options(bool plugin_options) {
         res = new (thd->mem_root)
             Item_int(iter->value.c_str(), (uint)iter->value.length());
         break;
-      case SHOW_CHAR:
       case SHOW_LEX_STRING:
-      case SHOW_BOOL:
-      case SHOW_MY_BOOL:
         res = new (thd->mem_root) Item_string(
             iter->value.c_str(), iter->value.length(), &my_charset_utf8mb4_bin);
         break;
+      case SHOW_CHAR: {
```

Comment:
Is my understanding correct that this part is needed for proper handling of wrong values saved previously in mysqld-auto.cnf file?
Having this patch all bools are saved as ON/OFF and enums are saved as their string representation (not corresponding int representation).

If the above is true, do we need this part of the patch? Isn't it enough to have fix for storing part?

Reply:
While working on this change I noticed that some incorrect values could still stay in config file. that is why I left this part. But for a resulting change you are probably right, there is not much sense in this code. Removed it.